microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/delay/ReadinessChecker.java

```diff
@@ -51,20 +51,20 @@ public Boolean call() {
                         if (response.getStatusCode() == 503) {
                             if (checks == null) {
                                 addState(DOWN_NO_CONTENT());
-                            } else if (checks.size() == 0) {
+                            } else if ((checks.size() == 4) && contains(checks, "empty-readiness-checks")) {
                                 addState(DOWN_NO_CHECK());
-                            } else if (checks.get(0).get("name").equals(DelayedReadinessHealthCheck.NAME)) {
+                            } else if ((checks.size() == 4) && contains(checks, DelayedReadinessHealthCheck.NAME)) {
```

Comment:
Can we verify new probes here? running test with parallel (separate terminal) `while true; do echo $(date +"%T.%3N") ;  curl   localhost:19990/health/ready; echo ""; done` produces:
```
...
===>>server does not exists
17:23:48.141
curl: (7) Failed to connect to localhost port 19990: Connection refused
===>>server is starting
17:23:48.154
{"status":"DOWN","checks":[{"name":"deployments-status","status":"DOWN"},{"name":"empty-readiness-checks","status":"DOWN"},{"name":"boot-errors","status":"DOWN"},{"name":"server-state","status":"DOWN"}]}
...
===>>deployment readiness probes are installed, deployment not ready
17:23:56.073
{"status":"DOWN","checks":[{"name":"deployments-status","status":"DOWN"},{"name":"boot-errors","status":"DOWN"},{"name":"server-state","status":"DOWN"},{"name":"delayed-readiness","status":"DOWN"}]}
===>>deployment readiness probes are installed, deployment ready
17:23:59.094
{"status":"DOWN","checks":[{"name":"deployments-status","status":"UP","data":{"NotReadyHealthCheckTest.war":"OK"}},{"name":"boot-errors","status":"UP"},{"name":"server-state","status":"UP","data":{"value":"running"}},{"name":"delayed-readiness","status":"DOWN"}]}
```

IMO we can increase test coverage here. Even for `UP` states.

But I don't know test plan and so on...So not sure if it not out of scope of the PR.

Reply:
Sorry I do not get your comment. fully. Where does `===>>` comes from? Do you see some problem while doing manual tests?

I must admit understanding and changing this class was for me tricky. Maybe we can arrange short call to sync.

Reply:
As discussed `===>>` is added by me to describe server state at the moment.

The main purpose of the test case is to verify that server is not ready (`UP`) before deployment probe is installed and would return `DOWN` from the begging. The scenario is still valid and the RFE does not change it. After discussion we have agreed that test coverage in `DefaultReadinessProcedureHealthTest` is sufficient. 

Thus the discussion has been resolved.